### ISSUE TITLE
Report unused direct dependencies for binary executables

### ISSUE THREAD

**Issue by scop** [rpm-software-management/rpmlint#5]
Currently unused direct dependencies are only reported for shared libraries.  It seems like they should be reported for binary executables as well.

Original comment by: opoplawski

**Comment by hobbes1069** [comment]
There are a lot... I just checked my system with the following one-liner:
```
for binary in $(ls /usr/bin/*); do echo "-- " $(basename $binary) " --"; ldd -u -r $binary; done > unused.txt
```

I would like to see this added as a warning so we can catch them as part of package reviews.

**Comment by marxin** [comment]
Let's implement that as a warning then. I see ~120 executables the have an unused dependency on my system.
